steps/generate_dataset_from_htme.py

```diff
@@ -0,0 +1,236 @@
+    for collection_dict in list_of_dicts:
+        for collection_name, collection_files_keys in collection_dict.items():
+            if collection_name.lower() in secrets_collections:
+                the_logger.info("Processing collection : " + collection_name)
+                tag_value = secrets_collections[collection_name.lower()]
+                print(f'Processing Collection {collection_name}')
+                initial_rdd = spark.sparkContext.emptyRDD()
+                for collection_file_key in collection_files_keys:
+                    encrypted = read_binary(f"s3://{s3_htme_bucket}/{collection_file_key}")
+                    metadata = get_metadatafor_key(collection_file_key)
+                    ciphertext = metadata["ciphertext"]
+                    datakeyencryptionkeyid = metadata["datakeyencryptionkeyid"]
+                    iv = metadata["iv"]
+                    plain_text_key = get_plaintext_key_calling_dks(ciphertext,datakeyencryptionkeyid)
+                    decrypted = encrypted.mapValues(lambda val: decrypt(plain_text_key,iv,val))
+                    decompressed = decrypted.mapValues(decompress)
+                    initial_rdd = initial_rdd.union(decompressed)
+                decoded_rdd = initial_rdd.mapValues(lambda txt: txt.decode("utf-8"))
+                row = Row("val")
+                decoded_df = decoded_rdd.flatMap(lambda x: x[1].split("\n")).map(row).toDF()
+                #decoded_df.show()
+                the_logger.info(f'No of records in collection : {collection_name} is  {decoded_df.count()}')
+                print(f'No of records in collection : {collection_name} is  {decoded_df.count()}')
+                the_logger.info("Persisting Json : " + collection_name)
+                json_location = persist_json(collection_name,decoded_df)
+                prefix = (
+                        "${file_location}/"
+                        + collection_name
+                        + "/"
+                        + collection_name
+                        + ".json"
+                )
+                the_logger.info("Applying Tags for prefix : " + prefix)
+                tag_objects(prefix,tag_value)
+                the_logger.info("Creating Hive tables for : " + collection_name)
+                create_hive_on_published(json_location, collection_name)
+            else:
+                logging.error(collection_name + " is not present in the collections list ",)
+                print(collection_name + " is not present in the collections list ")
+
+def get_metadatafor_key(key):
+    s3_object = s3_client.get_object(Bucket=s3_htme_bucket, Key=key)
+    #print(s3_object)
+    iv  = s3_object["Metadata"]["iv"]
+    ciphertext = s3_object["Metadata"]["ciphertext"]
+    datakeyencryptionkeyid = s3_object["Metadata"]["datakeyencryptionkeyid"]
+    #print ( "iv " + iv + "ciphertext" + ciphertext + "datakeyencryptionkeyid " + datakeyencryptionkeyid )
```

Comment:
Can we remove this commented out line please?

Reply:
> removed commented line